### PR TITLE
Add ASA scans running on CentOS 8

### PR DESCRIPTION
**Pull request by danielszot**
## Description

* ASA Scans running on CentOS 8 Added.
* Base container images locked by SHA256 to improve hermeticity.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [ ] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: devops/scripts/asa/run_asa.sh [hunk 1]
```diff
@@ -8,16 +8,33 @@ set -e
 # Tested and supported runtimes: podman, docker
 CONTAINER_RUNTIME="podman"
```
Comment (AhmedBM) [inline devops/scripts/asa/run_asa.sh:9]: can we have this resolved at runtime? eg. Use whichever is present podman/docker and report error if none?

**Review by AhmedBM — APPROVED** [review]
Just realized the container runtime definition wasn't even part of the change. Would be a great nice-to-have though

File: devops/scripts/asa/run_asa.sh (hunk 1 above)
Comment (AhmedBM) [reply devops/scripts/asa/run_asa.sh:9]: sorry just realized this wasn't even part of this changeset, im just thinking out loud here 😆

Comment (danielszot) [reply devops/scripts/asa/run_asa.sh:9]: Hehe, that's totally fine! By that I seen that we have podman set as default and it was supposed to be docker. I'll change that and will ask for re-approval.

I'll do runtime detection with priority to the first on the list in the follow up PR. That's not time consuming and rather straight forward.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Comment by danielszot** [comment]
Failing check seems not related to the PR itself.
It was also failing in the last merged PR (https://github.com/Azure/azure-osconfig/pull/813)
I think this one is rather done. @AhmedBM @MariusNi can you merge it?

**Pull request merged**